featuretools/primitives/standard/rolling_transform_primitive.py

```diff
@@ -421,13 +425,80 @@ def _pandas_std(series):
                 additional_args = (self.gap, _pandas_std, self.min_periods)
 
                 return rolled_series.apply(
-                    _apply_roll_with_offset_gap, args=additional_args
+                    apply_roll_with_offset_gap,
+                    args=additional_args,
                 ).values
             return rolled_series.std().values
 
         return rolling_std
 
 
+class RollingTrend(TransformPrimitive):
+    """Calculates the trend of a given window of entries of a column over time.
+
+    Description:
+        Given a list of numbers and a corresponding list of
+        datetimes, return a rolling slope of the linear trend
+        of values, starting at the row `gap` rows away from the current row and looking backward
+        over the specified time window (by `window_length` and `gap`).
+
+        Input datetimes should be monotonic.
+
+    Examples:
+        >>> import pandas as pd
+        >>> rolling_trend = RollingTrend()
+        >>> times = pd.date_range(start="2019-01-01", freq="1D", periods=10)
+        >>> rolling_trend(times, [1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, nan, 1.5, 3, 6, 8, 16, 36, 72, 144]
+
+        We can also control the gap before the rolling calculation.
+
+        >>> rolling_trend = RollingTrend(gap=1)
+        >>> rolling_trend(times, [1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, nan, nan, 3, 6, 8, 16, 36, 72]
+
+        We can also control the minimum number of periods required for the rolling calculation.
+
+        >>> rolling_trend = RollingTrend(window_length=4, min_periods=4)
+        >>> rolling_trend(times, [[1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, nan, 1.5, 2.3, 4.6, 6.8, 12.8, 26.4, 55.2,110.4]
+
+        We can also set the window_length and gap using offset alias strings.
+
+        >>> rolling_trend = RollingTrend(window_length="4D", gap="1D")
+        >>> rolling_trend(times, [[1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, 1, 1.5, 2.33333333, 3.75, 7.5, 13, 24, 46, 90])
+    """
+
+    name = "rolling_trend"
+    input_types = [
+        ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ColumnSchema(semantic_tags={"numeric"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+
+    def __init__(self, window_length=3, gap=0, min_periods=0):
+        self.window_length = window_length
+        self.gap = gap
+        self.min_periods = min_periods
+
+    def get_function(self):
+        def rolling_trend(datetime, numeric):
+            x = pd.Series(numeric.values, index=datetime.values)
+            rolled_series = roll_series_with_gap(
+                x, self.window_length, gap=self.gap, min_periods=self.min_periods
+            )
+            if isinstance(self.gap, str):
+                additional_args = (self.gap, np.mean, self.min_periods)
```

Comment:
`(self.gap, np.mean, self.min_periods)` is still calculating the average when an offset string gap is passed in. Should be `(self.gap, calculate_trend, self.min_periods)`

Reply:
oof